### PR TITLE
Snapshot/clone initial rpc

### PR DESCRIPTION
**Pull request by tiagolobocastro**
fix(snapshot/clone): implement create & list clone dataplane rpc api
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    feat(snapshot/clone): add scaffolding for dataplane rpc
    
    todo: add actual gRPC call implementation and type conversion.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

File: control-plane/stor-port/src/types/v0/transport/snapshot.rs [hunk 1]
```diff
@@ -120,3 +123,47 @@ pub enum ListReplicaSnapshots {
...
     Snapshot(SnapshotId),
 }
+
+/// The request type to create a snapshot's clone.
+pub type CreateSnapshotClone = SnapshotCloneParameters;
+
+/// Common set of parameters used for snapshot clone creation.
+#[derive(Debug, Clone)]
+pub struct SnapshotCloneParameters {
+    /// Unique identification of the snapshot source.
```
Comment (Abhinandan-Purkait) [inline control-plane/stor-port/src/types/v0/transport/snapshot.rs]: souce snapshot? Rather than snapshot source, as it indicates the volume I feel.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/grpc/proto/v1/replica/replica.proto [hunk 2]
```diff
@@ -59,6 +61,16 @@ enum ReplicaStatus {
   Faulted = 3;
 }
 
+// Kind of the Replica.
+enum ReplicaKind {
+  // A regular data-replica.
+  Regular = 0;
+  // A replica's snapshot, which is a replica.
+  Snapshot = 1;
+  // A replica's snapshot's clone, which is a replica.
```
Comment (dsharma-dc) [inline control-plane/grpc/proto/v1/replica/replica.proto]: How about `A replica that's cloned from some replica snapshot `?

File: control-plane/grpc/proto/v1/replica/replica.proto [hunk 3]
```diff
@@ -59,6 +61,16 @@ enum ReplicaStatus {
   Faulted = 3;
 }
 
+// Kind of the Replica.
+enum ReplicaKind {
+  // A regular data-replica.
+  Regular = 0;
+  // A replica's snapshot, which is a replica.
```
Comment (dsharma-dc) [inline control-plane/grpc/proto/v1/replica/replica.proto]: `A replica which is created as a snapshot of some replica.` ?

**Review by dsharma-dc — APPROVED** [review]

File: control-plane/stor-port/src/types/v0/transport/snapshot.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/stor-port/src/types/v0/transport/snapshot.rs]: Done

File: control-plane/grpc/proto/v1/replica/replica.proto (hunk 3 above)
Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/replica/replica.proto]: Tweaked it a bit:
    /// A replica which is another replica's snapshot.
    /// A replica which is a clone of another replica's snapshot.

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**